flytekit/types/pickle/pickle.py

```diff
@@ -52,31 +52,31 @@ def python_type(cls) -> typing.Type:
         return _SpecificFormatClass
 
     @classmethod
-    def to_pickle(cls, ctx: FlyteContext, python_val: typing.Any) -> str:
+    async def to_pickle(cls, ctx: FlyteContext, python_val: typing.Any) -> str:
         local_dir = ctx.file_access.get_random_local_directory()
         os.makedirs(local_dir, exist_ok=True)
         local_path = ctx.file_access.get_random_local_path()
         uri = os.path.join(local_dir, local_path)
         with open(uri, "w+b") as outfile:
             cloudpickle.dump(python_val, outfile)
 
-        return ctx.file_access.put_raw_data(uri)
+        return await ctx.file_access.async_put_raw_data(uri)
 
     @classmethod
-    def from_pickle(cls, uri: str) -> typing.Any:
+    async def from_pickle(cls, uri: str) -> typing.Any:
```

Comment:
@pingsutw are these public facing functions?

Reply:
not public facing